iceoryx_posh/test/moduletests/test_shared_memory_user.cpp

```diff
@@ -0,0 +1,106 @@
+
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using namespace iox::roudi;
+
+namespace iox
+{
+namespace test
+{
+class SharedMemoryUserTest : public Test
+{
+  public:
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    const uint64_t segmentId{1U};
+    rp::BaseRelativePointer::offset_t segmentManagerAddressOffset{0U};
+    void SetUp(){}
+    void TearDown(){}
+};
+
+TEST_F(SharedMemoryUserTest, ConstructorShmObjectWithZeroTopicSizeAndDoMapSharedMemoryIntoThreadBoolSetToTrueReturnError)
+{
+    iox::cxx::optional<iox::Error> detectedError;
+    const bool doMapSharedMemoryIntoThread{"TRUE"};
+    const size_t topicSize{0U};
+
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+    [&detectedError](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+    detectedError.emplace(error);
+    EXPECT_EQ(errorLevel, iox::ErrorLevel::FATAL);
+    });
+    SharedMemoryUser ShmUser(doMapSharedMemoryIntoThread, topicSize, segmentId, segmentManagerAddressOffset);
+  
+    ASSERT_TRUE(detectedError.has_value());
+    EXPECT_EQ(detectedError.value(), iox::Error::kPOSH__SHM_APP_MAPP_ERR);
+}
+
+TEST_F(SharedMemoryUserTest, ConstructorShmObjectWithZeroTopicSizeAndDoMapSharedMemoryIntoThreadBoolSetToFalseReturnError)
+{
+    iox::cxx::optional<iox::Error> detectedError;
+    const bool doMapSharedMemoryIntoThread{"FALSE"};
+    const size_t topicSize{0U};
+
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+    [&detectedError](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+    detectedError.emplace(error);
+    EXPECT_EQ(errorLevel, iox::ErrorLevel::FATAL);
+    });
+    SharedMemoryUser ShmUser(doMapSharedMemoryIntoThread, topicSize, segmentId, segmentManagerAddressOffset);
+  
+    ASSERT_TRUE(detectedError.has_value());
+    EXPECT_EQ(detectedError.value(), iox::Error::kPOSH__SHM_APP_MAPP_ERR);
+}
+
+TEST_F(SharedMemoryUserTest, ConstructorShmObjectWithValidTopicSizeAndDoMapSharedMemoryIntoThreadBoolSetToFalseReturnNoError)
+{
+    iox::cxx::optional<iox::Error> detectedError;
+    const bool doMapSharedMemoryIntoThread{"FALSE"};
+    const size_t topicSize{1U};
+
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+    [&detectedError](const iox::Error error, const std::function<void()>, const iox::ErrorLevel errorLevel) {
+    detectedError.emplace(error);
+    EXPECT_EQ(errorLevel, iox::ErrorLevel::FATAL);
+    });
+
+    SharedMemoryUser ShmUser(doMapSharedMemoryIntoThread, topicSize, segmentId, segmentManagerAddressOffset);
+    EXPECT_FALSE(detectedError.has_value());
+}
+
+TEST_F(SharedMemoryUserTest, ConstructorShmObjectWithValidTopicSizeAndDoMapSharedMemoryIntoThreadBoolSetToTrueReturnNoError)
+{
+    iox::cxx::optional<iox::Error> detectedError;
+    const bool doMapSharedMemoryIntoThread{"TRUE"};
```

Comment:
There are tests missing which verify that the shared memory was actually mapped into the process. The opposite can maybe tested with a death test.